File: frontend/src/app/models/project.ts
export type ProjectLanguage = 'Python';
export type ProjectModel = 'GPT-4o';
export type ProjectAgent = 'BasicAgent' | 'L3-Reasoning' | 'Undefined';

export interface Project {
  id: string;
  name: string;
  language: ProjectLanguage;
  model: ProjectModel;
  updatedAt: Date;
  agent: ProjectAgent;
}

File: backend/functions/models/project.py
from datetime import datetime
from typing import Literal, TypedDict


ProjectLanguage = Literal["Python"]
ProjectModel = Literal["GPT-4o"]


class Project(TypedDict):
    """
    Projects are JSON objects stored as dictionaries.
    """

    id: str
    name: str
    language: ProjectLanguage
    model: ProjectModel
    updatedAt: datetime
